### DISCUSSION TITLE
Typescript compatibility with Fast Design/Web Components?

### DISCUSSION THREAD

**Discussion by kotx** [solidjs/solid#1432] [Q&A]
Microsoft's [Fast Design](https://fast.design) exposes UI components as Web Components. However with Typescript all Web Components fail typechecks with `Property 'my-element' does not exist on type 'JSX.IntrinsicElements'.ts(2339)`
I can work around this with this in `element.d.ts`:
```ts
import "solid-js"

declare module "solid-js" {
	namespace JSX {
		interface IntrinsicElements {
			"fast-button": any
		}
	}
}
```

But then I lose type inference/etc for `fast-button`. I also have to do this for every `fast-*` element/Web Component I want to use. 

Is it possible to either disable this check entirely or have proper types for all `fast-*` components? I don't want to do this for every component I want to use.

**Comment by atk** [comment]
We should definitely add this to our documentation: https://github.com/solidjs/solid/issues/616#issuecomment-1144074821

**Reply by kotx** [reply]
I saw that earlier, but it seems FAST Design doesn't add its components to `HTMLElementTagNameMap`. I suppose that's an issue on their end that I should open, but it's not ideal.

**Comment by atk** [comment]
Does FAST expose any types?

**Reply by kotx** [reply]
I'm not too sure of any that would help, but there's an [`allComponents`](https://www.fast.design/docs/api/fast-components.allcomponents) export.
